azurerm/internal/services/cosmos/common/schema.go

```diff
@@ -37,3 +40,56 @@ func MongoCollectionAutoscaleSettingsSchema() *schema.Schema {
 
 	return autoscaleSettingsDatabaseSchema
 }
+
+func CosmosDbIndexingPolicySchema() *schema.Schema {
+	return &schema.Schema{
+		Type:     schema.TypeList,
```

Comment:
Is there a suggested pattern for validating `TypeList` that does cross field validation? In the indexing policy case there are the validations of

- `included_path` and `excluded_path` can't be set when the `indexing_mode` is `None`
- One of `included_path` or `excluded_path` must contain the path `/*`

Reply:
The best solution is to validate it in create and return an error at that point if it fails